src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/Tensor.cs

```diff
@@ -3653,7 +3591,9 @@ public static ref readonly TensorSpan<T> DegreesToRadians<T>(scoped in ReadOnlyT
         public static T Distance<T>(scoped in ReadOnlyTensorSpan<T> x, scoped in ReadOnlyTensorSpan<T> y)
             where T : IRootFunctions<T>
         {
-            return TensorPrimitivesHelperTwoSpanInTOut(x, y, TensorPrimitives.Distance);
+            TensorOperation.ValidateCompatibility(x, y);
+            Distance<T> op = default;
+            return TensorOperation.Invoke<TensorOperation.Distance<T>, T, T>(x, y, op);
```

Comment:
This could use `SumSquaredDifferences` and then take the square root, right? No need to carry state around.